api/apps/api/src/modules/geo-features/processing/split-events.handler.ts

```diff
@@ -0,0 +1,45 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { API_EVENT_KINDS } from '@marxan/api-events';
+import { FeaturesJobData } from '@marxan/geofeature-calculations';
+import { CreateApiEventDTO } from '@marxan-api/modules/api-events/dto/create.api-event.dto';
+import {
+  CreateWithEventFactory,
+  EventData,
+  EventFactory,
+  QueueEventsAdapter,
+} from '@marxan-api/modules/queue-api-events';
+import { splitEventsFactoryToken } from './queue-providers';
+
+@Injectable()
+export class SplitEventsHandler implements EventFactory<FeaturesJobData> {
```

Comment:
It's handling events of the queue by creating the `queueEvents` and binding DTO creation methods for the mapping. Additionally, it may listen on `queueEvents` to react on not-duplicated reemitted events